### PR TITLE
Build Devise invitable UI

### PR DESCRIPTION
**Pull request by JoshDevHub**
## This PR
- Generates views/mailers for `devise_invitable`
- Styles new forms in accordance with existing login page styles
- Adds an `InvitationPolicy` and uses it to restrict creating invitations to admin users. This policy is also tested.
- Adds letter opener

I ended up doing a couple of things a bit differently than the design doc. One is that actually using/testing this email system is much easier with letter opener, so I just went ahead and added it in. 

The other is that I was originally going to use the builtin controller filter `devise_invitable` has to lock sending invitations to admins only. But that builtin filter is looking at class names to make its determinations, and our app has all users under one class with the `role` field differentiating members and admins. I'm sure I could customize this, but I also wanted to conditionally display a link in the navbar for sending out invitations. At that point, I figured I'd just make a Pundit policy for it and handle the authorization through that.

Maybe one of things I'd specifically like feedback on is:
```rb
class InvitationsController < Devise::InvitationsController
```
Not sure about my controller name here, but a better name didn't immediately occur to me. `ApplicationInvitationsController` maybe?

### CODE REVIEW

**Comment by toyhammered** [comment]
> Not sure about my controller name here, but a better name didn't immediately occur to me. `ApplicationInvitationsController` maybe?

That is perfect to me.

File: spec/policies/invitation_policy_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    it 'denies access to regular member users' do
+      expect(subject).not_to permit(member)
+    end
+
+    it 'grants access to admin users' do
+      expect(subject).to permit(admin)
+    end
+  end
+
+  permissions :edit?, :update? do
```
Comment (toyhammered) [inline spec/policies/invitation_policy_spec.rb:19]: What would be a use case for this? Also shouldn't it be scoped to that user directly can only edit/update their own?

**Review by toyhammered — COMMENTED** [review]
This looks great!

File: spec/policies/invitation_policy_spec.rb (hunk 1 above)
Comment (JoshDevHub) [reply spec/policies/invitation_policy_spec.rb:19]: This is kind of due to the way the gem works, I was initially a bit puzzled by it.

If you look at `app/views/devise/invitations/edit.html.erb` you'll see what essentially looks like a sign up page. What happens is a user is created with their email address when the invite is initially sent. When the user clicks on the invite link in their email, they're taken to that "edit" page to enter their password (and potentially `timezone`, etc. in the future) and of course `update` is what happens when they submit.

This flow is already fully locked down by the invitation token. No user can visit the edit page or submit a response to it without a valid invitation token. So I just put `true` there to keep pundit out of the way on that. It's already tightly controlled by the `devise_invitations` gem itself.

Comment (toyhammered) [reply spec/policies/invitation_policy_spec.rb:19]: ahhh perfect. Thank you for this explanation.

**Review by toyhammered — APPROVED** [review]

File: spec/policies/invitation_policy_spec.rb (hunk 1 above)
Comment (GALTdea) [reply spec/policies/invitation_policy_spec.rb:19]: Cool, I'll have this flow  in mind when implementing timezone

**Pull request merged**